src/Swarm/Util/RunCode.hs

```diff
@@ -0,0 +1,28 @@
+module Swarm.Util.RunCode where
+
+import Control.Arrow (left)
+import Control.Monad.Except
+import Data.Text (Text)
+import Data.Text qualified as T (pack)
+import Data.Text.IO qualified as TIO
+import Swarm.Language.Pipeline (ProcessedTerm, processTermEither)
+
+data SolutionSource
+  = ScenarioSuggested
+  | PlayerAuthored
+
+data CodeToRun = CodeToRun SolutionSource ProcessedTerm
+
+getParsedInitialCode :: Maybe FilePath -> ExceptT Text IO (Maybe CodeToRun)
```

Comment:
This function can not return ScenarioSuggested solution.

Please do not create Util modules for things that belong somewhere.

When you want to parse a term, please put that function next to other functions that do so.

When you create a data type used in scenarios and initialising game state, please put it near game state.

Reply:
I agree, the `Util` namespace should be used only for things that get used from many unrelated places in the code.